### PR TITLE
Release/3.2.0

### PR DESCRIPTION
**Pull request by mpu-creare**
### Features
* Added the "none" interpolator, which disables the Interpolation node of Mixin
* Enabled nearest-neighbor interpolation for N-D stacked coordinates (#488)
* Implemented Xarray interpolation for dependent coordinates (multi-dimensional stacked coordinates)
* Improved caching
  * Extra diemnsions are discarded when caching datasource and compositor outputs
  * Added CRS-agnostic caching
* Added utility function that generates a JSON spec of available PODPAC nodes. This spec can be used to make UI elements
* Added the `get_source_data` method for `Datasource` nodes to bypass interpolation (#485)
* Added the `get_bounds` method to `Node` to retrieve the coordinate bounds (#486)
* Added `Node` point prober features that will evaluate every node in a pipeline at a point and record its value. Useful for debugging complex pipelines.
* Added `AffineCoorindates` (#491) these are stacked spatial shaped coordinates that are parameterized by a `geotransform`. Replaces `RotatedCoordinates`.
* `OGC.WCS` `Node` can now handle multi-band geotiffs

### Bugfixes
* Fixed a number of floating point coordinates discrepancies
* `Rasterio` node will no longer attempt to use overviews when no overviews are present.
* Fixed `geotransform` shape transpose error
* Fixed numpy deprecation of using `list` indices
* Fixed regressions due to dependency updates

### CODE REVIEW

**Comment by mpu-creare** [comment]
@jmilloy it's been a while since we've made a release. I just wanted a quick sanity check from another set of eyes. Most of the changes happened months ago and were reviewed as part of other PRs. 

Thanks for asking!

**Review by jmilloy — APPROVED** [review]
Yes, this is a lot stuff that we worked through thoroughly before merging. I found a typo in stacked coordinates that must not come up much by running the tests, and ran the black autoformatting.

**Comment by mpu-creare** [comment]
Thanks a lot @jmilloy:
*  I'm surprised the formatting was that off -- my git hook prevents me from checking in anything without the formatting
* Thanks a lot for catching that typo -- would have been a huge pain at some point no doubt. I'm also surprised the unit tests did not catch that. 

Merging...

**Pull request merged**
